File: Ahorn/entities/AxisMinion.jl
module AxisMinion

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/AxisMinion" AxisMinion(x::Integer, y::Integer, detectionRange::Integer=120, health::Integer=2, moveSpeed::Integer=60, patrolDistance::Integer=80)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(AxisMinion)
)

function Ahorn.selection(entity::AxisMinion)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::AxisMinion, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/axisMinion.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local axisMinion = {}

axisMinion.name = "MaggyHelper/AxisMinion"
axisMinion.depth = -100
axisMinion.fieldInformation = {
    health = {
        minimumValue = 1
    },
    moveSpeed = {
        minimumValue = 0
    },
    detectionRange = {
        minimumValue = 0
    },
    patrolDistance = {
        minimumValue = 0
    }
}
axisMinion.placements = {
    {
        name = "normal",
        data = {
            health = 2,
            moveSpeed = 60,
            detectionRange = 120,
            patrolDistance = 80
        }
    }
}

function axisMinion.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/axis_minion/idle", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function axisMinion.selection(room, entity)
    return utils.rectangle(entity.x - 10, entity.y - 24, 20, 24)
end

return axisMinion
